tss-esapi/src/context/tpm_commands/duplication_commands.rs

```diff
@@ -331,5 +333,356 @@ impl Context {
+    /// #     )
+    /// #     .expect("tr_sess_set_attributes call failed");
+    /// #
+    /// # let policy_session = PolicySession::try_from(policy_auth_session)
+    /// #     .expect("Failed to convert auth session into policy session");
+    /// #
+    /// # context
+    /// #     .policy_auth_value(policy_session)
+    /// #     .expect("Policy auth value");
+    /// #
+    /// # context
+    /// #     .policy_command_code(policy_session, TPM2_CC_Duplicate)
+    /// #     .unwrap();
+    /// #
+    /// # context.set_sessions((Some(policy_auth_session), None, None));
+    /// #
+    /// # let (encryption_key_out, duplicate, out_sym_seed) = context
+    /// #     .duplicate(
+    /// #         object_to_duplicate_handle,
+    /// #         new_parent_handle,
+    /// #         None,
+    /// #         SymmetricDefinitionObject::Null,
+    /// #     )
+    /// #     .unwrap();
+    /// # eprintln!("D: {:?}, P: {:?}, S: {:?}", encryption_key_out, duplicate, out_sym_seed);
+    /// # let public = context.read_public(object_to_duplicate_handle.into()).unwrap().0;
+    /// #
+    /// # let session = context
+    /// #     .start_auth_session(
+    /// #         None,
+    /// #         None,
+    /// #         None,
+    /// #         SessionType::Hmac,
+    /// #         SymmetricDefinition::AES_256_CFB,
+    /// #         HashingAlgorithm::Sha256,
+    /// #     )
+    /// #     .unwrap();
+    /// # let (session_attributes, session_attributes_mask) = SessionAttributesBuilder::new()
+    /// #     .with_decrypt(true)
+    /// #     .with_encrypt(true)
+    /// #     .build();
+    /// # context.tr_sess_set_attributes(
+    /// #     session.unwrap(),
+    /// #     session_attributes,
+    /// #     session_attributes_mask,
+    /// # )
+    /// # .unwrap();
+    /// # context.set_sessions((session, None, None));
+    ///
+    /// // `encryption_key_out`, `duplicate` and `out_sym_seed` are generated
+    /// // by `duplicate` function
+    /// let private = context.import(
+    ///     new_parent_handle,
+    ///     Some(encryption_key_out),
+    ///     public,
+    ///     duplicate,
+    ///     out_sym_seed,
+    ///     SymmetricDefinitionObject::Null,
+    ///  ).unwrap();
+    /// #
+    /// # eprintln!("P: {:?}", private);
+    /// ```
+    pub fn import(
+        &mut self,
+        parent_handle: ObjectHandle,
+        encryption_key: Option<Data>,
+        public: Public,
+        duplicate: Private,
+        encrypted_secret: EncryptedSecret,
+        symmetric_alg: SymmetricDefinitionObject,
+    ) -> Result<Private> {
+        let mut out_private = null_mut();
+        let ret = unsafe {
+            Esys_Import(
+                self.mut_context(),
+                parent_handle.into(),
+                self.required_session_1()?,
+                self.optional_session_2(),
+                self.optional_session_3(),
+                &encryption_key.unwrap_or_default().into(),
```

Comment:
Hmm, following [the discussion on Slack](https://cloud-native.slack.com/archives/C01EARH2ZB3/p1633333906080300), maybe this should be documented (by this I mean what the result is of using `None` for `encryption_key`)

Reply:
Yep :+1: I'll do that for import and duplicate in a separate commit in a moment.

Browsing through the code it seems `unwrap_or_default` appears heavily in [`create_primary`](https://docs.rs/tss-esapi/7.0.0-alpha.1/src/tss_esapi/context/tpm_commands/hierarchy_commands.rs.html#30-96). I'm thinking through @Superhepper's ["explicit default"](https://cloud-native.slack.com/archives/C01EARH2ZB3/p1633340711082700?thread_ts=1633333906.080300&cid=C01EARH2ZB3) and yeah, if writing `Default::default()` was as short as `None` I wouldn't see a reason to prefer `None`. (it seems `Default::default` [cannot be imported directly](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0012f4cb66c28b48a6aa43a57f2c0991), but having `None` and `Some(Default::default())` mean the same seems redundant at best).